{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d973c785-541f-4d65-9095-51dbb644dda3",
   "metadata": {},
   "source": [
    "# load packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b7d65bc-215c-489b-a92d-ddd459449f72",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ca27eb5-3ab2-4795-a043-053255d939fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import StandardScaler, MinMaxScaler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af905dcf-aa94-4791-8391-df6577080090",
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50421ddb-b709-41bf-bbc4-4762a5166d72",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "263650f1-a88e-4aeb-9f55-3f49d6778cb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d51d5a9-5e3a-41ad-8217-238a9595d285",
   "metadata": {},
   "source": [
    "# read in input files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09a94c26-8c5f-4a7e-994e-8e7e6519f29c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_score_adsp = pd.read_csv('common_var_gene_score/igap_adsp_gene_score/merged_outputs/AOU_ALL.UKBB.metasoft.ADSP.all.VEP_v113.gene_by_position.r2_0.1_clump_variants_excluded.RE_pval_threshold_0.05.gene_symbol.average_gene_score.merged.common_id.transpose.txt.gz',\n",
    "                              sep = '\\t')\n",
    "print(gene_score_adsp.shape)\n",
    "print(gene_score_adsp[~gene_score_adsp['GENE'].isin(['ALZ_STATUS', 'AGE', 'SEX', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8'])].drop(columns = ['GENE']).astype(float).min().min())\n",
    "print(gene_score_adsp[~gene_score_adsp['GENE'].isin(['ALZ_STATUS', 'AGE', 'SEX', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8'])].drop(columns = ['GENE']).astype(float).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9d82751-4b0d-4011-b638-a7f632a5bfe8",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_rosmap = pd.read_csv('pathway_score/gene_mapping/ROSMAP.RNAseq.TPM5_log2norm.individualids.codinggenes.VEP_v113_genes.gene_symbol.common_id.duplicate_ids_removed.transpose.txt.gz',\n",
    "                            sep = '\\t')\n",
    "print(rnaseq_rosmap.drop(columns = ['GENE']).min().min())\n",
    "print(rnaseq_rosmap.drop(columns = ['GENE']).max().max())\n",
    "print(rnaseq_rosmap.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5956e589-bd1b-407d-8a6c-de3474046334",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_10_msbb = pd.read_csv('pathway_score/msbb/MSBB.RNAseq.BA_10.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.txt',\n",
    "                             sep = '\\t')\n",
    "print(rnaseq_10_msbb.shape)\n",
    "print(rnaseq_10_msbb.drop(columns = ['GENE']).min().min())\n",
    "print(rnaseq_10_msbb.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff788bd0-ea1b-4b1e-ae3d-b25f812470c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_22_msbb = pd.read_csv('pathway_score/msbb/MSBB.RNAseq.BA_22.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.txt',\n",
    "                             sep = '\\t')\n",
    "print(rnaseq_22_msbb.shape)\n",
    "print(rnaseq_22_msbb.drop(columns = ['GENE']).min().min())\n",
    "print(rnaseq_22_msbb.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97501a4a-cfc5-46c0-9b6f-56708372d6ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_36_msbb = pd.read_csv('pathway_score/msbb/MSBB.RNAseq.BA_36.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.txt',\n",
    "                             sep = '\\t')\n",
    "print(rnaseq_36_msbb.shape)\n",
    "print(rnaseq_36_msbb.drop(columns = ['GENE']).min().min())\n",
    "print(rnaseq_36_msbb.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf3bc736-572b-4cc4-ba5f-f92f237b6354",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_44_msbb = pd.read_csv('pathway_score/msbb/MSBB.RNAseq.BA_44.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.txt',\n",
    "                             sep = '\\t')\n",
    "print(rnaseq_44_msbb.shape)\n",
    "print(rnaseq_44_msbb.drop(columns = ['GENE']).min().min())\n",
    "print(rnaseq_44_msbb.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4adcbc72-a2c4-47de-9fed-f970c1ff056a",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_rosmap = pd.read_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.txt',\n",
    "                            sep = '\\t')\n",
    "print(methyl_rosmap.shape)\n",
    "print(methyl_rosmap.drop(columns = ['GENE']).min().min())\n",
    "print(methyl_rosmap.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5ff1e40-5cdc-4452-bb5a-3fa7568ad27b",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_msbb = pd.read_csv('pathway_score/msbb/MSBB.methylation_array.19batch.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.txt',\n",
    "                          sep = '\\t')\n",
    "print(methyl_msbb.shape)\n",
    "print(methyl_msbb.drop(columns = ['GENE']).min().min())\n",
    "print(methyl_msbb.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48adc759-b36e-4f55-aa45-e2ee53442c24",
   "metadata": {},
   "outputs": [],
   "source": [
    "somoscan_rosmap = pd.read_csv('pathway_score/rosmap/ROSMAP.proteomics.somoscan.individualID.entrez_gene_symbol.common_id.transpose.txt.gz',\n",
    "                              sep = '\\t')\n",
    "print(somoscan_rosmap.shape)\n",
    "print(somoscan_rosmap.drop(columns = ['GENE']).min().min())\n",
    "print(somoscan_rosmap.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd2ae433-85c0-4b13-8e16-3a480d9338ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmt_msbb = pd.read_csv('pathway_score/msbb/MSBB.TMT_proteomics.19batch.normalized.gene_symbol.individualID.log2_transformed.common_id.transpose.txt',\n",
    "                       sep = '\\t')\n",
    "print(tmt_msbb.shape)\n",
    "print(tmt_msbb.drop(columns = ['GENE']).min().min())\n",
    "print(tmt_msbb.drop(columns = ['GENE']).max().max())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74d3be7f-4645-41e6-8d0a-b3bf179e7f5e",
   "metadata": {},
   "source": [
    "# clean up dfs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ddc31ed-ee03-4297-a041-2f0ba140448e",
   "metadata": {},
   "source": [
    "## gene scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e97511c7-bee8-44ed-8f01-d75b786c40ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_score_adsp_clean = gene_score_adsp[~gene_score_adsp['GENE'].isin(['ALZ_STATUS', 'AGE', 'SEX', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8'])]\n",
    "gene_score_adsp_clean.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0adcb201-5ceb-44e9-9c85-ab6d516a44bd",
   "metadata": {},
   "source": [
    "## replace inf with missing in methlation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22fca2c3-e355-4c70-94f9-cb9543e392b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_rosmap_clean = methyl_rosmap.replace([np.inf, -np.inf], np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94b521e6-8700-4f05-82bf-70f5784f3d25",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_msbb_clean = methyl_msbb.replace([np.inf, -np.inf], np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b67935d0-63ae-4e72-b6d9-a6d5fd392c38",
   "metadata": {},
   "source": [
    "# scale"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9485f548-531e-447f-9683-5ce877e11873",
   "metadata": {},
   "source": [
    "## gene score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c26893a6-9051-4f7f-b5a2-b03f907f45f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = gene_score_adsp_clean.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "gene_score_adsp_scaled_df = output_df.copy()\n",
    "gene_score_adsp_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a42fd1af-df27-428f-af92-cdf8d84fbb20",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = gene_score_adsp_clean.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "gene_score_adsp_minmax_scaled_df = output_df.copy()\n",
    "gene_score_adsp_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80823e56-78ac-45d0-8f4a-43d38880bda1",
   "metadata": {},
   "source": [
    "## rosmap rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1df05cf0-e900-4548-8ad2-c243bf52f7ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_rosmap.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "rnaseq_rosmap_scaled_df = output_df.copy()\n",
    "rnaseq_rosmap_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1675ac1-44a6-4f63-9540-35667c570c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_rosmap.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "rnaseq_rosmap_minmax_scaled_df = output_df.copy()\n",
    "rnaseq_rosmap_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c72ee872-366c-4e34-b224-71bcf60ccd19",
   "metadata": {},
   "source": [
    "## msbb rnaseq 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44435f79-d6d5-4377-8267-4f7e928757fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_10_msbb.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "rnaseq_10_msbb_scaled_df = output_df.copy()\n",
    "rnaseq_10_msbb_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35ce0581-7bd9-4890-be9a-ff2b47284327",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_10_msbb.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "rnaseq_10_msbb_minmax_scaled_df = output_df.copy()\n",
    "rnaseq_10_msbb_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9122c1d9-74fb-4dbf-be62-c7be1707d447",
   "metadata": {},
   "source": [
    "## msbb rnaseq 22"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71a08203-a5c5-43c1-8d6f-b8196e149a54",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_22_msbb.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "rnaseq_22_msbb_scaled_df = output_df.copy()\n",
    "rnaseq_22_msbb_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0b45202-2158-46c6-9364-62f370abfeb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_22_msbb.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "rnaseq_22_msbb_minmax_scaled_df = output_df.copy()\n",
    "rnaseq_22_msbb_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b5478d1-e040-4b5e-8533-2046a094cbdb",
   "metadata": {},
   "source": [
    "## msbb rnaseq 36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56f530ee-cd45-45ce-83ae-0ac746099054",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_36_msbb.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "rnaseq_36_msbb_scaled_df = output_df.copy()\n",
    "rnaseq_36_msbb_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5781cc0-e001-4825-95c3-a3cd111f2835",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_36_msbb.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "rnaseq_36_msbb_minmax_scaled_df = output_df.copy()\n",
    "rnaseq_36_msbb_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "548ef3e0-0868-4292-a059-7541736e668d",
   "metadata": {},
   "source": [
    "## msbb rnaseq 44"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90c781dc-4cb5-4a76-ad14-7da13d5f012a",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_44_msbb.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "rnaseq_44_msbb_scaled_df = output_df.copy()\n",
    "rnaseq_44_msbb_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a00e24a-2e70-41a1-ac62-a13632d5505b",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = rnaseq_44_msbb.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "rnaseq_44_msbb_minmax_scaled_df = output_df.copy()\n",
    "rnaseq_44_msbb_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d46761e-ea02-4b4e-9394-0bdaed2a02c1",
   "metadata": {},
   "source": [
    "## methyl rosmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "384c87e2-fef3-46c0-acbc-7fd94648ceaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = methyl_rosmap_clean.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "methyl_rosmap_scaled_df = output_df.copy()\n",
    "print(methyl_rosmap_scaled_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5162bfe1-1db8-4a5e-bd6d-9a46dee73a21",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = methyl_rosmap_clean.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "methyl_rosmap_minmax_scaled_df = output_df.copy()\n",
    "methyl_rosmap_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b716043b-c9d9-4a1b-9e4c-d1dcade21017",
   "metadata": {},
   "source": [
    "## methyl msbb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7409ba2-8f01-401a-a486-c6f3b221b35f",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = methyl_msbb_clean.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "methyl_msbb_scaled_df = output_df.copy()\n",
    "print(methyl_msbb_scaled_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "605a21c7-fa13-4a9b-b4a7-542445884e0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = methyl_msbb_clean.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "methyl_msbb_minmax_scaled_df = output_df.copy()\n",
    "methyl_msbb_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5941c61-e29f-4277-b3a4-854b73eca578",
   "metadata": {},
   "source": [
    "## rosmap somoscan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba475b3a-d86b-4e35-bd73-5e0b86cbaef9",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = somoscan_rosmap.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "somoscan_rosmap_scaled_df = output_df.copy()\n",
    "somoscan_rosmap_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e143bfc-441d-47bc-bb57-3a1074722983",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = somoscan_rosmap.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "somoscan_rosmap_minmax_scaled_df = output_df.copy()\n",
    "somoscan_rosmap_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "753e190c-8dd3-41d4-b014-048e90f701ed",
   "metadata": {},
   "source": [
    "## tmt msbb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52521817-f9e0-4917-8c36-43122b1bffde",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = tmt_msbb.copy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].mean().mean())\n",
    "print(input_df[predictors].std().mean())\n",
    "\n",
    "print(output_df[predictors].mean().mean())\n",
    "print(output_df[predictors].std().mean())\n",
    "\n",
    "tmt_msbb_scaled_df = output_df.copy()\n",
    "tmt_msbb_scaled_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ecab60ae-338e-46e5-bdb5-93491a0da19d",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_df = tmt_msbb.copy()\n",
    "\n",
    "scaler = MinMaxScaler(feature_range = (-14.472073470708509, 28.78601705422015))\n",
    "predictors = [col for col in input_df.columns if col != 'GENE']\n",
    "gene = input_df[['GENE']]\n",
    "output_df = scaler.fit_transform(input_df[predictors])\n",
    "output_df = pd.DataFrame(output_df)\n",
    "output_df = pd.concat([gene, output_df], axis = 1)\n",
    "output_df.columns = input_df.columns\n",
    "\n",
    "print(input_df[predictors].min().min())\n",
    "print(input_df[predictors].max().max())\n",
    "\n",
    "print(output_df[predictors].min().min())\n",
    "print(output_df[predictors].max().max())\n",
    "\n",
    "tmt_msbb_minmax_scaled_df = output_df.copy()\n",
    "tmt_msbb_minmax_scaled_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8e0a557-ad75-4ec0-a1b0-4b0112308cc0",
   "metadata": {},
   "source": [
    "# export"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cad96812-81a3-4533-83a2-9cd1a1fb0b6c",
   "metadata": {},
   "source": [
    "## gene score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "824deaad-4444-491e-9f80-40df04e76648",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_score_adsp_scaled_df.to_csv('common_var_gene_score/igap_adsp_gene_score/merged_outputs/AOU_ALL.UKBB.metasoft.ADSP.all.VEP_v113.gene_by_position.r2_0.1_clump_variants_excluded.RE_pval_threshold_0.05.gene_symbol.average_gene_score.merged.common_id.transpose.scaled.txt',\n",
    "                                 sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f7f9822-25d6-400f-b3c5-bff0f32e8b69",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_score_adsp_minmax_scaled_df.to_csv('common_var_gene_score/igap_adsp_gene_score/merged_outputs/AOU_ALL.UKBB.metasoft.ADSP.all.VEP_v113.gene_by_position.r2_0.1_clump_variants_excluded.RE_pval_threshold_0.05.gene_symbol.average_gene_score.merged.common_id.transpose.minmax.scaled.txt',\n",
    "                                        sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e956c42b-e5c0-45b0-aaa2-0953e8ec52c4",
   "metadata": {},
   "source": [
    "## rosmap rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c442f286-51e7-4355-9ed4-0cc5b9dee5bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_rosmap_scaled_df.to_csv('pathway_score/gene_mapping/ROSMAP.RNAseq.TPM5_log2norm.individualids.codinggenes.VEP_v113_genes.gene_symbol.common_id.duplicate_ids_removed.transpose.scaled.txt',\n",
    "                               sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd5c2bb0-f265-4b61-9437-2017e5660c6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_rosmap_minmax_scaled_df.to_csv('pathway_score/gene_mapping/ROSMAP.RNAseq.TPM5_log2norm.individualids.codinggenes.VEP_v113_genes.gene_symbol.common_id.duplicate_ids_removed.transpose.minmax.scaled.txt',\n",
    "                                      sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f111b7d1-da17-4298-a8be-34d0848114c3",
   "metadata": {},
   "source": [
    "## msbb 10 rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1fcc4e1-4231-4132-9cdb-9c54efd5d43f",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_10_msbb_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_10.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt',\n",
    "                                sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d2f2ab2-8899-43e8-bc5d-cb522b15bbc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_10_msbb_minmax_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_10.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.minmax.scaled.txt',\n",
    "                                       sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8832168b-138a-407b-86a8-e865341d3fec",
   "metadata": {},
   "source": [
    "## msbb 22 rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c67f12b-e7ad-4099-b141-7956580f1fd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_22_msbb_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_22.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt',\n",
    "                                sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2921a703-005e-4375-bf62-7108cf33367e",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_22_msbb_minmax_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_22.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.minmax.scaled.txt',\n",
    "                                       sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "857a5b92-a4eb-4848-9789-387d93027266",
   "metadata": {},
   "source": [
    "## msbb 36 rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b12e902-6948-49e9-865c-6ea15ce721eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_36_msbb_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_36.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt',\n",
    "                                sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b3abf0c-c09f-4a65-b7e9-ccd8b412bf69",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_36_msbb_minmax_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_36.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.minmax.scaled.txt',\n",
    "                                       sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9dffba18-bd33-4808-af68-65207d4a807e",
   "metadata": {},
   "source": [
    "## msbb 44 rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17797843-35e7-43a1-813d-9473da2a6bd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_44_msbb_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_44.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt',\n",
    "                                sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0e55a40-3dc6-46c0-a1bd-06a6b8584bc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_44_msbb_minmax_scaled_df.to_csv('pathway_score/msbb/MSBB.RNAseq.BA_44.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.minmax.scaled.txt',\n",
    "                                       sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e44f357b-b4b3-4d6d-9cc0-b176d5395db6",
   "metadata": {},
   "source": [
    "## methyl rosmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5449e6a2-1117-4d7b-9e3c-a8ebde569d5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_rosmap_scaled_df.to_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.scaled.txt',\n",
    "                               sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "392325ac-6dad-4ecb-ac74-1727e65b77ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_rosmap_minmax_scaled_df.to_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.minmax.scaled.txt',\n",
    "                                      sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "966f01c8-aaba-4109-b66d-5c22ad204bd2",
   "metadata": {},
   "source": [
    "## methyl msbb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1eeecbf2-574c-47a2-86c9-bd816bdaf7eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_msbb_scaled_df.to_csv('pathway_score/msbb/MSBB.methylation_array.19batch.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.scaled.txt',\n",
    "                             sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9476aea8-07ed-4e9c-a128-3dea1cb80616",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_msbb_minmax_scaled_df.to_csv('pathway_score/msbb/MSBB.methylation_array.19batch.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.minmax.scaled.txt',\n",
    "                                    sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5355f0a7-891e-452b-b9c0-192160de4ff4",
   "metadata": {},
   "source": [
    "## somoscan rosmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "feab3588-4697-4e28-ad23-96c951a1d255",
   "metadata": {},
   "outputs": [],
   "source": [
    "somoscan_rosmap_scaled_df.to_csv('pathway_score/rosmap/ROSMAP.proteomics.somoscan.individualID.entrez_gene_symbol.common_id.transpose.scaled.txt',\n",
    "                                 sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba45e85e-e08a-43bc-a70f-44fb8d32a0d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "somoscan_rosmap_minmax_scaled_df.to_csv('pathway_score/rosmap/ROSMAP.proteomics.somoscan.individualID.entrez_gene_symbol.common_id.transpose.minmax.scaled.txt',\n",
    "                                        sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08aaf4a7-0fe3-4a49-a632-c3801140a580",
   "metadata": {},
   "source": [
    "## tmt msbb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a34715d-3505-4c58-bbb2-c7f502033f36",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmt_msbb_scaled_df.to_csv('pathway_score/msbb/MSBB.TMT_proteomics.19batch.normalized.gene_symbol.individualID.log2_transformed.common_id.transpose.scaled.txt',\n",
    "                          sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c128aa68-201b-475f-a0c3-e527452154c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmt_msbb_minmax_scaled_df.to_csv('pathway_score/msbb/MSBB.TMT_proteomics.19batch.normalized.gene_symbol.individualID.log2_transformed.common_id.transpose.minmax.scaled.txt',\n",
    "                                 sep = '\\t', index = None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.12 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
