{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "056202c6-345f-4296-bb07-d4cf33e2d528",
   "metadata": {},
   "source": [
    "# load packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d88daa10-c209-4194-8c8a-c0be715e6071",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.metrics import roc_auc_score, average_precision_score, f1_score, balanced_accuracy_score\n",
    "import numpy as np\n",
    "import xgboost as xgb"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36177f9f-5439-437e-a8e9-ab95cc14fdaa",
   "metadata": {},
   "source": [
    "# read in input files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de87d483-fda9-45f7-95f2-101ade47a9d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_score_adsp = pd.read_csv(('common_var_gene_score/igap_adsp_gene_score/merged_outputs/AOU_ALL.UKBB.metasoft.ADSP.all.VEP_v113.gene_by_position.r2_0.1_clump_variants_excluded.RE_pval_threshold_0.05.gene_symbol.average_gene_score.merged.common_id.transpose.scaled.txt.gz'),\n",
    "                              sep = '\\t', low_memory = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44b06241-a21a-462d-a170-be3897fe2610",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_rosmap = pd.read_csv(('pathway_score/gene_mapping/ROSMAP.RNAseq.TPM5_log2norm.individualids.codinggenes.VEP_v113_genes.gene_symbol.common_id.duplicate_ids_removed.transpose.scaled.txt'),\n",
    "                            sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "942310fe-49d7-46eb-b5d5-6f42be9ba75c",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_10_msbb = pd.read_csv(('pathway_score/msbb/MSBB.RNAseq.BA_10.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt'),\n",
    "                             sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0e75f45-00d9-4b5d-845a-4a17daa5a299",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_22_msbb = pd.read_csv(('pathway_score/msbb/MSBB.RNAseq.BA_22.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt'),\n",
    "                             sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b91b3fe-db5a-4bbc-bac0-035bf6c560fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_36_msbb = pd.read_csv(('pathway_score/msbb/MSBB.RNAseq.BA_36.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt'),\n",
    "                             sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf363b09-960f-49cd-b7e4-39d14262f622",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_44_msbb = pd.read_csv(('pathway_score/msbb/MSBB.RNAseq.BA_44.19batch.gene_symbol.individualID.mvalue_norm.tpm.log2.coding_genes_only.covar_corrected.common_id.transpose.scaled.txt'),\n",
    "                             sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d92ba86-0e93-49f8-9cfd-bd56842ed17f",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_rosmap = pd.read_csv(('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.scaled.txt'),\n",
    "                            sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "330851ba-ac3a-42e8-9e26-baefa7bd661b",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_msbb = pd.read_csv(('pathway_score/msbb/MSBB.methylation_array.19batch.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.scaled.txt'),\n",
    "                          sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a6f8c4b-c48f-4e4a-a04f-1a5b6c85a4d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "somoscan_rosmap = pd.read_csv(('pathway_score/rosmap/ROSMAP.proteomics.somoscan.individualID.entrez_gene_symbol.common_id.transpose.scaled.txt'),\n",
    "                              sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "323bda03-3914-431c-a7c5-7165d0b645bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmt_msbb = pd.read_csv(('pathway_score/msbb/MSBB.TMT_proteomics.19batch.normalized.gene_symbol.individualID.log2_transformed.common_id.transpose.scaled.txt'),\n",
    "                       sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13d04e35-bb44-41e6-ae13-8139de64329a",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_path_map = pd.read_csv(('pathway_score/pathway_annotation/go/AD_KMI.ADSP.ROSMAP.all_omics.MSBB.all_omics.VEP_113.ref_gene.go.gene_to_pathway.no_duplicates.pathway_mapping.txt'),\n",
    "                            sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ad0bee47-6dfa-423a-910e-4bab10d194b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "id_map_keep_quest_comb = pd.read_csv('pathway_score/id_map/ADSP.ROSMAP.MSBB.keep_quest_comb.id_map.txt',\n",
    "                                     sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf3d3936-7ff1-4c42-99fb-fadac683437d",
   "metadata": {},
   "outputs": [],
   "source": [
    "adsp_pheno = pd.read_csv('/project/ritchie/projects/ADSP_Projects/QC_ADSPv11/ADSPphenotype_forAnalysis.txt',\n",
    "                         sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f91fb7a-cdd4-415e-bd0c-a46dbdead6e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "id_map = pd.read_csv('pathway_score/id_map/ADSP.ROSMAP.MSBB.id_map.txt',\n",
    "                      sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18316596-4078-4d63-af06-b453ab123404",
   "metadata": {},
   "outputs": [],
   "source": [
    "best_model = pd.read_csv('ML/statistical_models/xgboost_output/ALL_SPLITS.MEAN_METRICS.XGBoost.avg_pathway.standard_scaled.go.keep_quest_comb.csv')\n",
    "best_model.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "087a7d0c-92ab-42bb-bacb-4f2955c8b48c",
   "metadata": {},
   "source": [
    "# clean files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9f856a2-7675-4f43-be6d-3d486304b27c",
   "metadata": {},
   "outputs": [],
   "source": [
    "adsp_pheno = adsp_pheno.rename(columns = {'IID' : 'SampleID'})\n",
    "adsp_pheno_id = id_map.merge(adsp_pheno, on = 'SampleID', how = 'inner')\n",
    "adsp_pheno_id = adsp_pheno_id[adsp_pheno_id['SampleID'].isin(id_map_keep_quest_comb['SampleID'])]\n",
    "adsp_pheno_id = adsp_pheno_id[['CommonID', 'DX_harmonized', 'Age_harmonized', 'Sex', 'PC1', 'PC2', 'PC3', 'PC4']]\n",
    "adsp_pheno_id.rename(columns = {'CommonID' : 'ID',\n",
    "                                'DX_harmonized' : 'AD',\n",
    "                                'Age_harmonized' : 'Age'}, inplace = True)\n",
    "adsp_pheno_id = adsp_pheno_id.set_index('ID', drop = True)\n",
    "print(len(adsp_pheno_id.index))\n",
    "adsp_pheno_id.dropna(inplace = True)\n",
    "print(len(adsp_pheno_id.index))\n",
    "adsp_pheno_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3dbefbd4-3bd1-4ba8-9686-00fb0f8b45b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_score_adsp = gene_score_adsp[~gene_score_adsp['GENE'].isin(['ALZ_STATUS', 'AGE', 'SEX', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8'])]\n",
    "gene_score_adsp = gene_score_adsp[gene_score_adsp['GENE'].isin(all_path_map['GENE'])]\n",
    "gene_score_adsp.set_index('GENE', inplace = True)\n",
    "gene_score_adsp = gene_score_adsp.transpose()\n",
    "gene_score_adsp = gene_score_adsp.join(adsp_pheno_id, how = 'inner')\n",
    "gene_score_adsp = gene_score_adsp.dropna(axis = 1, how = 'all')\n",
    "print(gene_score_adsp.shape)\n",
    "gene_score_adsp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e0716df-f561-4eb2-b6c7-a4bf44aa83ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_rosmap = rnaseq_rosmap[rnaseq_rosmap['GENE'].isin(all_path_map['GENE'])]\n",
    "rnaseq_rosmap.set_index('GENE', inplace = True)\n",
    "rnaseq_rosmap = rnaseq_rosmap.transpose()\n",
    "rnaseq_rosmap = rnaseq_rosmap.join(adsp_pheno_id, how = 'inner')\n",
    "rnaseq_rosmap = rnaseq_rosmap.dropna(axis = 1, how = 'all')\n",
    "print(rnaseq_rosmap.shape)\n",
    "rnaseq_rosmap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0d94d75-394a-4f0f-87a2-a26628a48ca5",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_10_msbb = rnaseq_10_msbb[rnaseq_10_msbb['GENE'].isin(all_path_map['GENE'])]\n",
    "rnaseq_10_msbb.set_index('GENE', inplace = True)\n",
    "rnaseq_10_msbb = rnaseq_10_msbb.transpose()\n",
    "rnaseq_10_msbb = rnaseq_10_msbb.join(adsp_pheno_id, how = 'inner')\n",
    "rnaseq_10_msbb = rnaseq_10_msbb.dropna(axis = 1, how = 'all')\n",
    "print(rnaseq_10_msbb.shape)\n",
    "rnaseq_10_msbb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "535bc688-1f3f-44ad-b075-5f7f16a41654",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_22_msbb = rnaseq_22_msbb[rnaseq_22_msbb['GENE'].isin(all_path_map['GENE'])]\n",
    "rnaseq_22_msbb.set_index('GENE', inplace = True)\n",
    "rnaseq_22_msbb = rnaseq_22_msbb.transpose()\n",
    "rnaseq_22_msbb = rnaseq_22_msbb.join(adsp_pheno_id, how = 'inner')\n",
    "rnaseq_22_msbb = rnaseq_22_msbb.dropna(axis = 1, how = 'all')\n",
    "print(rnaseq_22_msbb.shape)\n",
    "rnaseq_22_msbb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4639533b-e998-41e5-a477-2a90e40987d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_36_msbb = rnaseq_36_msbb[rnaseq_36_msbb['GENE'].isin(all_path_map['GENE'])]\n",
    "rnaseq_36_msbb.set_index('GENE', inplace = True)\n",
    "rnaseq_36_msbb = rnaseq_36_msbb.transpose()\n",
    "rnaseq_36_msbb = rnaseq_36_msbb.join(adsp_pheno_id, how = 'inner')\n",
    "rnaseq_36_msbb = rnaseq_36_msbb.dropna(axis = 1, how = 'all')\n",
    "print(rnaseq_36_msbb.shape)\n",
    "rnaseq_36_msbb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f49697f5-e73b-40ef-a8c4-5ecee9c8835f",
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_44_msbb = rnaseq_44_msbb[rnaseq_44_msbb['GENE'].isin(all_path_map['GENE'])]\n",
    "rnaseq_44_msbb.set_index('GENE', inplace = True)\n",
    "rnaseq_44_msbb = rnaseq_44_msbb.transpose()\n",
    "rnaseq_44_msbb = rnaseq_44_msbb.join(adsp_pheno_id, how = 'inner')\n",
    "rnaseq_44_msbb = rnaseq_44_msbb.dropna(axis = 1, how = 'all')\n",
    "print(rnaseq_44_msbb.shape)\n",
    "rnaseq_44_msbb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4956f2e4-0969-4d99-8ffe-d21f06e8116e",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_rosmap = methyl_rosmap[methyl_rosmap['GENE'].isin(all_path_map['GENE'])]\n",
    "methyl_rosmap.set_index('GENE', inplace = True)\n",
    "methyl_rosmap = methyl_rosmap.transpose()\n",
    "methyl_rosmap = methyl_rosmap.join(adsp_pheno_id, how = 'inner')\n",
    "methyl_rosmap = methyl_rosmap.dropna(axis = 1, how = 'all')\n",
    "print(methyl_rosmap.shape)\n",
    "methyl_rosmap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7afc0dca-e52d-4686-8042-253392a718e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_msbb = methyl_msbb[methyl_msbb['GENE'].isin(all_path_map['GENE'])]\n",
    "methyl_msbb.set_index('GENE', inplace = True)\n",
    "methyl_msbb = methyl_msbb.transpose()\n",
    "methyl_msbb = methyl_msbb.join(adsp_pheno_id, how = 'inner')\n",
    "methyl_msbb = methyl_msbb.dropna(axis = 1, how = 'all')\n",
    "print(methyl_msbb.shape)\n",
    "methyl_msbb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e09a3b31-b8f3-4cd0-8c84-9bd5b2bc5146",
   "metadata": {},
   "outputs": [],
   "source": [
    "somoscan_rosmap = somoscan_rosmap[somoscan_rosmap['GENE'].isin(all_path_map['GENE'])]\n",
    "somoscan_rosmap.set_index('GENE', inplace = True)\n",
    "somoscan_rosmap = somoscan_rosmap.transpose()\n",
    "somoscan_rosmap = somoscan_rosmap.join(adsp_pheno_id, how = 'inner')\n",
    "somoscan_rosmap = somoscan_rosmap.dropna(axis = 1, how = 'all')\n",
    "print(somoscan_rosmap.shape)\n",
    "somoscan_rosmap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80b09051-7543-4196-9267-35482e428005",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmt_msbb = tmt_msbb[tmt_msbb['GENE'].isin(all_path_map['GENE'])]\n",
    "tmt_msbb.set_index('GENE', inplace = True)\n",
    "tmt_msbb = tmt_msbb.transpose()\n",
    "tmt_msbb = tmt_msbb.join(adsp_pheno_id, how = 'inner')\n",
    "tmt_msbb = tmt_msbb.dropna(axis = 1, how = 'all')\n",
    "print(tmt_msbb.shape)\n",
    "tmt_msbb.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db3c10f2-b066-49e1-abc7-9a37cff8b883",
   "metadata": {},
   "source": [
    "# run tests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fce4933-5d6f-428c-91ef-cb3921ab6c1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = gene_score_adsp.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "gs_adsp_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "gs_adsp_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "gs_adsp_f1 = f1_score(y_test, y_pred_bin)\n",
    "gs_adsp_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "gs_adsp_num = len(df.index)\n",
    "\n",
    "print(gs_adsp_auroc)\n",
    "print(gs_adsp_auprc)\n",
    "print(gs_adsp_f1)\n",
    "print(gs_adsp_balanced_acc)\n",
    "print(gs_adsp_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f5f831d-877b-4c70-9206-e2c42ee75f23",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = rnaseq_rosmap.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "rs_rosmap_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "rs_rosmap_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "rs_rosmap_f1 = f1_score(y_test, y_pred_bin)\n",
    "rs_rosmap_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "rs_rosmap_num = len(df.index)\n",
    "\n",
    "print(rs_rosmap_auroc)\n",
    "print(rs_rosmap_auprc)\n",
    "print(rs_rosmap_f1)\n",
    "print(rs_rosmap_balanced_acc)\n",
    "print(rs_rosmap_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e20621f7-c282-4132-aa80-41bb38fb02c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = rnaseq_10_msbb.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "rs_10_msbb_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "rs_10_msbb_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "rs_10_msbb_f1 = f1_score(y_test, y_pred_bin)\n",
    "rs_10_msbb_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "rs_10_msbb_num = len(df.index)\n",
    "\n",
    "print(rs_10_msbb_auroc)\n",
    "print(rs_10_msbb_auprc)\n",
    "print(rs_10_msbb_f1)\n",
    "print(rs_10_msbb_balanced_acc)\n",
    "print(rs_10_msbb_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f917a59-b16e-415f-ba64-a1ab5335a806",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = rnaseq_22_msbb.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "rs_22_msbb_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "rs_22_msbb_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "rs_22_msbb_f1 = f1_score(y_test, y_pred_bin)\n",
    "rs_22_msbb_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "rs_22_msbb_num = len(df.index)\n",
    "\n",
    "print(rs_22_msbb_auroc)\n",
    "print(rs_22_msbb_auprc)\n",
    "print(rs_22_msbb_f1)\n",
    "print(rs_22_msbb_balanced_acc)\n",
    "print(rs_22_msbb_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8f39a7b-f86a-489b-895a-92869d9a5212",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = rnaseq_36_msbb.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "rs_36_msbb_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "rs_36_msbb_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "rs_36_msbb_f1 = f1_score(y_test, y_pred_bin)\n",
    "rs_36_msbb_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "rs_36_msbb_num = len(df.index)\n",
    "\n",
    "print(rs_36_msbb_auroc)\n",
    "print(rs_36_msbb_auprc)\n",
    "print(rs_36_msbb_f1)\n",
    "print(rs_36_msbb_balanced_acc)\n",
    "print(rs_36_msbb_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7eb0abb7-79b4-492e-b9bd-ea7b066fa4f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = rnaseq_44_msbb.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "rs_44_msbb_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "rs_44_msbb_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "rs_44_msbb_f1 = f1_score(y_test, y_pred_bin)\n",
    "rs_44_msbb_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "rs_44_msbb_num = len(df.index)\n",
    "\n",
    "print(rs_44_msbb_auroc)\n",
    "print(rs_44_msbb_auprc)\n",
    "print(rs_44_msbb_f1)\n",
    "print(rs_44_msbb_balanced_acc)\n",
    "print(rs_44_msbb_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df0b2a17-526e-4eb0-b3e3-d3c17856c628",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = methyl_rosmap.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "methyl_rosmap_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "methyl_rosmap_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "methyl_rosmap_f1 = f1_score(y_test, y_pred_bin)\n",
    "methyl_rosmap_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "methyl_rosmap_num = len(df.index)\n",
    "\n",
    "print(methyl_rosmap_auroc)\n",
    "print(methyl_rosmap_auprc)\n",
    "print(methyl_rosmap_f1)\n",
    "print(methyl_rosmap_balanced_acc)\n",
    "print(methyl_rosmap_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1b5281aa-e436-4efb-9cde-48d5391d3eab",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = methyl_msbb.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "methyl_msbb_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "methyl_msbb_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "methyl_msbb_f1 = f1_score(y_test, y_pred_bin)\n",
    "methyl_msbb_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "methyl_msbb_num = len(df.index)\n",
    "\n",
    "print(methyl_msbb_auroc)\n",
    "print(methyl_msbb_auprc)\n",
    "print(methyl_msbb_f1)\n",
    "print(methyl_msbb_balanced_acc)\n",
    "print(methyl_msbb_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "637724a1-6a6f-436d-abca-07fe1031b591",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = somoscan_rosmap.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "somoscan_rosmap_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "somoscan_rosmap_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "somoscan_rosmap_f1 = f1_score(y_test, y_pred_bin)\n",
    "somoscan_rosmap_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "somoscan_rosmap_num = len(df.index)\n",
    "\n",
    "print(somoscan_rosmap_auroc)\n",
    "print(somoscan_rosmap_auprc)\n",
    "print(somoscan_rosmap_f1)\n",
    "print(somoscan_rosmap_balanced_acc)\n",
    "print(somoscan_rosmap_num)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ee9f154-7d37-4c06-9599-601097414532",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = tmt_msbb.copy()\n",
    "\n",
    "train = df.sample(frac = 0.7, random_state = 7)\n",
    "test = df.drop(train.index)\n",
    "\n",
    "x_train = train.drop(columns = ['AD'])\n",
    "y_train = train[['AD']]\n",
    "x_test = test.drop(columns = ['AD'])\n",
    "y_test = test[['AD']]\n",
    "\n",
    "model = xgb.XGBClassifier(objective = 'binary:logistic', random_state = 7, n_jobs = -1)\n",
    "model.fit(x_train, y_train)\n",
    "\n",
    "y_pred_bin = model.predict(x_test)\n",
    "y_pred_cont = model.predict_proba(x_test)[:, 1]\n",
    "\n",
    "tmt_msbb_auroc = roc_auc_score(y_test, y_pred_cont)\n",
    "tmt_msbb_auprc = average_precision_score(y_test, y_pred_cont)\n",
    "tmt_msbb_f1 = f1_score(y_test, y_pred_bin)\n",
    "tmt_msbb_balanced_acc = balanced_accuracy_score(y_test, y_pred_bin)\n",
    "tmt_msbb_num = len(df.index)\n",
    "\n",
    "print(tmt_msbb_auroc)\n",
    "print(tmt_msbb_auprc)\n",
    "print(tmt_msbb_f1)\n",
    "print(tmt_msbb_balanced_acc)\n",
    "print(tmt_msbb_num)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0ae787a-787f-4c7b-8e82-b57e9ca4070d",
   "metadata": {},
   "source": [
    "# extract best model metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "daf51d1c-8a07-437e-8c14-778ee6a4a0b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "best_model_auroc = best_model.loc[0, 'TEST_AUROC_MEAN']\n",
    "best_model_auprc = best_model.loc[0, 'TEST_AUPRC_MEAN']\n",
    "best_model_f1 = best_model.loc[0, 'TEST_F1_MEAN']\n",
    "best_model_balanced_acc = best_model.loc[0, 'TEST_BALANCED_ACCURACY_MEAN']\n",
    "best_model_num = 17123\n",
    "\n",
    "print(best_model_auroc)\n",
    "print(best_model_auprc)\n",
    "print(best_model_f1)\n",
    "print(best_model_balanced_acc)\n",
    "print(best_model_num)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23b4452c-941a-4688-8e0a-19c1c901d203",
   "metadata": {},
   "source": [
    "# make combined df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab70a1e2-b597-4c62-b3f3-ca78f20acdec",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_metics = pd.DataFrame(data = {'METRIC' : ['AUROC', 'AUPRC', 'F1_SCORE', 'BALANCED_ACCURACY', 'N_SAMPLES'],\n",
    "                                  'BEST_PATHWAY_SCORE' : [best_model_auroc, best_model_auprc, best_model_f1, best_model_balanced_acc, best_model_num],\n",
    "                                  'GENE_SCORE_ADSP' : [gs_adsp_auroc, gs_adsp_auprc, gs_adsp_f1, gs_adsp_balanced_acc, gs_adsp_num],\n",
    "                                  'RNASEQ_ROSMAP' : [rs_rosmap_auroc, rs_rosmap_auprc, rs_rosmap_f1, rs_rosmap_balanced_acc, rs_rosmap_num],\n",
    "                                  'RNASEQ_BA10_MSBB' : [rs_10_msbb_auroc, rs_10_msbb_auprc, rs_10_msbb_f1, rs_10_msbb_balanced_acc, rs_10_msbb_num],\n",
    "                                  'RNASEQ_BA22_MSBB' : [rs_22_msbb_auroc, rs_22_msbb_auprc, rs_22_msbb_f1, rs_22_msbb_balanced_acc, rs_22_msbb_num],\n",
    "                                  'RNASEQ_BA36_MSBB' : [rs_36_msbb_auroc, rs_36_msbb_auprc, rs_36_msbb_f1, rs_36_msbb_balanced_acc, rs_36_msbb_num],\n",
    "                                  'RNASEQ_BA44_MSBB' : [rs_44_msbb_auroc, rs_44_msbb_auprc, rs_44_msbb_f1, rs_44_msbb_balanced_acc, rs_44_msbb_num],\n",
    "                                  'METHYL_ROSMAP' : [methyl_rosmap_auroc, methyl_rosmap_auprc, methyl_rosmap_f1, methyl_rosmap_balanced_acc, methyl_rosmap_num],\n",
    "                                  'METHYL_MSBB' : [methyl_msbb_auroc, methyl_msbb_auprc, methyl_msbb_f1, methyl_msbb_balanced_acc, methyl_msbb_num],\n",
    "                                  'SOMOSCAN_ROSMAP' : [somoscan_rosmap_auroc, somoscan_rosmap_auprc, somoscan_rosmap_f1, somoscan_rosmap_balanced_acc, somoscan_rosmap_num],\n",
    "                                  'TMT_MSBB' : [tmt_msbb_auroc, tmt_msbb_auprc, tmt_msbb_f1, tmt_msbb_balanced_acc, tmt_msbb_num]})\n",
    "all_metics"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb5dd126-aac4-4eee-a2fc-18ffd6787cba",
   "metadata": {},
   "source": [
    "# identify overlapping samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65f1efd1-0948-4421-9d3f-36f545815fc9",
   "metadata": {},
   "source": [
    "## make id lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a25da78-4e01-4982-b027-866e01e7c8ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "gs_adsp_unique_id = gene_score_adsp.reset_index()[['index']]\n",
    "gs_adsp_unique_id['SOURCE'] = 'GS_ADSP'\n",
    "gs_adsp_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb053303-79ce-4335-b5a9-68259ca8df14",
   "metadata": {},
   "outputs": [],
   "source": [
    "rs_rosmap_unique_id = rnaseq_rosmap.reset_index()[['index']]\n",
    "rs_rosmap_unique_id['SOURCE'] = 'RS_ROSMAP'\n",
    "rs_rosmap_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d1395d3-0a99-45ca-8dc3-2d87ea8a2f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "rs_10_msbb_unique_id = rnaseq_10_msbb.reset_index()[['index']]\n",
    "rs_10_msbb_unique_id['SOURCE'] = 'RS_10_MSBB'\n",
    "rs_10_msbb_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "caefcbf1-fd2d-4def-a9fe-87616f53386c",
   "metadata": {},
   "outputs": [],
   "source": [
    "rs_22_msbb_unique_id = rnaseq_22_msbb.reset_index()[['index']]\n",
    "rs_22_msbb_unique_id['SOURCE'] = 'RS_22_MSBB'\n",
    "rs_22_msbb_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ec2e452-d7ab-422e-8003-e8ca2a570255",
   "metadata": {},
   "outputs": [],
   "source": [
    "rs_36_msbb_unique_id = rnaseq_36_msbb.reset_index()[['index']]\n",
    "rs_36_msbb_unique_id['SOURCE'] = 'RS_36_MSBB'\n",
    "rs_36_msbb_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43e3a764-f572-48a8-bbf3-b1cd724c0815",
   "metadata": {},
   "outputs": [],
   "source": [
    "rs_44_msbb_unique_id = rnaseq_44_msbb.reset_index()[['index']]\n",
    "rs_44_msbb_unique_id['SOURCE'] = 'RS_44_MSBB'\n",
    "rs_44_msbb_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffe48eda-0b05-4818-b7c8-4a27dc6902b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ma_rosmap_unique_id = methyl_rosmap.reset_index()[['index']]\n",
    "ma_rosmap_unique_id['SOURCE'] = 'MA_ROSMAP'\n",
    "ma_rosmap_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66a96332-14a5-48ff-8efc-4d5f15469bd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "ma_msbb_unique_id = methyl_msbb.reset_index()[['index']]\n",
    "ma_msbb_unique_id['SOURCE'] = 'MA_MSBB'\n",
    "ma_msbb_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0dc7f741-df60-47ef-9ff3-d3a6f8496592",
   "metadata": {},
   "outputs": [],
   "source": [
    "sp_rosmap_unique_id = somoscan_rosmap.reset_index()[['index']]\n",
    "sp_rosmap_unique_id['SOURCE'] = 'SP_ROSMAP'\n",
    "sp_rosmap_unique_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d446619-5bad-4402-be8d-cdae75149c43",
   "metadata": {},
   "outputs": [],
   "source": [
    "tp_msbb_unique_id = tmt_msbb.reset_index()[['index']]\n",
    "tp_msbb_unique_id['SOURCE'] = 'TP_MSBB'\n",
    "tp_msbb_unique_id.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f839a942-17d6-4cd7-b2e2-32191c289a00",
   "metadata": {},
   "source": [
    "## concatenate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5821feed-a6f2-4ce1-b88d-11db6db9d6b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_ids = pd.concat([gs_adsp_unique_id,\n",
    "                     rs_rosmap_unique_id,\n",
    "                     rs_10_msbb_unique_id,\n",
    "                     rs_22_msbb_unique_id,\n",
    "                     rs_36_msbb_unique_id,\n",
    "                     rs_44_msbb_unique_id,\n",
    "                     ma_rosmap_unique_id,\n",
    "                     ma_msbb_unique_id,\n",
    "                     sp_rosmap_unique_id,\n",
    "                     tp_msbb_unique_id], axis = 0)\n",
    "print(len(all_ids.index))\n",
    "print(len(all_ids['index'].unique()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f87e51f-e0e0-4fea-beea-395730f3e21f",
   "metadata": {},
   "source": [
    "## extract duplicate ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3931ff73-4dde-40e9-8e3f-a458e53f44c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "dup_ids = all_ids[all_ids['index'].duplicated(keep = False)]\n",
    "print(len(dup_ids.index))\n",
    "print(len(dup_ids['index'].unique()))\n",
    "dup_ids['SOURCE'].value_counts(dropna = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ce19747-e0fd-4479-94aa-0f7e82883943",
   "metadata": {},
   "source": [
    "# export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5390599-064d-4063-b453-a85076aefd99",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_metics.to_csv('ML/statistical_models/indiv_omics/AD.all_omics.standard_scaled.model_prediction.csv', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b4ae3f8-2f4b-4c2d-a23f-b97ff49b5b13",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.12 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
