{
 "cells": [
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "0",
=======
   "id": "2c7b926b-0f1b-4778-b9ff-8b1ed5268d64",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "# load packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "1",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)"
=======
   "id": "b7a6a974-6ecb-455c-acea-285e1a97fa17",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "3",
=======
   "id": "59434d09-a69b-46fc-ab7b-e72b0ef78199",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "# read in input files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "4",
   "metadata": {},
   "outputs": [],
   "source": [
    "id_map = read.csv('rosmap/ID_mapping/ROSMAP.Methylation_Array.individualID_map.txt',\n",
    "                  sep = '\\t',\n",
    "                  na.strings = c(''))\n",
    "head(id_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5",
   "metadata": {},
   "outputs": [],
   "source": [
    "methylation = fread('rosmap/ROSMAP_arrayMethylation_imputed.tsv.gz',\n",
    "                    sep = '\\t',\n",
    "                    na.strings = c(''))\n",
    "head(methylation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map = read.csv('rosmap/ROSMAP_arrayMethylation_metaData.tsv',\n",
    "                    sep = '\\t',\n",
    "                    na.strings = c(''))\n",
    "head(gene_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref = fread('ensembl_start_stop/Homo_sapiens.GRCh38.113.refseq.tsv.gz',\n",
    "            sep = '\\t')\n",
    "head(ref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8",
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_methyl = fread('rosmap/Epigenetics/Epigenetics (DNA methylation array)/ROSMAP_arrayMethylation_raw.gz',\n",
    "                   sep = '\\t')\n",
    "raw_methyl.head()"
=======
   "id": "96fda289-a81a-4672-854f-eeab79869c47",
   "metadata": {},
   "outputs": [],
   "source": [
    "id_map = pd.read_csv('pathway_score/rosmap/ID_mapping/ROSMAP.Methylation_Array.individualID_map.txt',\n",
    "                     sep = '\\t')\n",
    "print(id_map.shape)\n",
    "id_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00eee4f5-2f53-4cad-83cb-34c0cffea131",
   "metadata": {},
   "outputs": [],
   "source": [
    "common_id_map = pd.read_csv('pathway_score/id_map/ADSP.ROSMAP.MSBB.id_map.txt',\n",
    "                            sep = '\\t')\n",
    "print(common_id_map.shape)\n",
    "common_id_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3efd1a9-a842-43d3-8362-84c79ca7f78c",
   "metadata": {},
   "outputs": [],
   "source": [
    "methylation = pd.read_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.tsv.gz',\n",
    "                          sep = '\\t')\n",
    "print(methylation.shape)\n",
    "methylation.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a53cb50-725e-41bc-a439-6693063ecac4",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map = pd.read_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_metaData.tsv',\n",
    "                       sep = '\\t')\n",
    "print(gene_map.shape)\n",
    "gene_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "582b5834-1e6c-42ea-b823-7effd3b0cf1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref = pd.read_csv('/project/ritchie/projects/ADSP_Projects/ADSP_Annotations/VEP_annotation_manual_113/ensembl_start_stop/Homo_sapiens.GRCh38.113.refseq.tsv.gz',\n",
    "                  sep = '\\t',\n",
    "                  dtype = {'RefGene' : str})\n",
    "print(ref.shape)\n",
    "ref.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "9",
   "metadata": {},
   "source": [
    "# look at data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(id_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(methylation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(gene_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map %>%\n",
    "filter(!is.na(RefGene)) %>%\n",
    "nrow()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(unique(methylation$TargetID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(unique(gene_map$TargetID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(unique(gene_map$RefGene))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17",
=======
   "id": "12cb7336-bb10-4b08-9d81-f43cc6545a9f",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "# clean up meta data"
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "18",
=======
   "id": "967eecc9-fc7c-43bb-a628-2b885f3fb86b",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## filter to non-missing ref genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "19",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map = gene_map %>%\n",
    "filter(!is.na(RefGene))\n",
    "dim(ref_gene_map)\n",
    "length(unique(ref_gene_map$RefGene))\n",
    "head(ref_gene_map)"
=======
   "id": "641fe5e5-d805-4f7f-9c6e-72d7a7199e93",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map = gene_map[~gene_map['RefGene'].isna()]\n",
    "print(ref_gene_map.shape)\n",
    "print(len(ref_gene_map['RefGene'].unique()))\n",
    "print(ref_gene_map['Island_Feature'].value_counts(dropna = False))\n",
    "ref_gene_map.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "20",
=======
   "id": "16992fbb-3ba7-477c-87af-7f369de08853",
   "metadata": {},
   "source": [
    "## explode semicolon separate values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0007a29c-8024-4659-ba66-c6849c34cfc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map['RefGene'] = ref_gene_map['RefGene'].str.split(';')\n",
    "ref_gene_map['Accession'] = ref_gene_map['Accession'].str.split(';')\n",
    "ref_gene_map['Gene_Feature'] = ref_gene_map['Gene_Feature'].str.split(';')\n",
    "ref_gene_map_explode = ref_gene_map.explode(['RefGene', 'Accession', 'Gene_Feature'])\n",
    "ref_gene_map_explode = ref_gene_map_explode.drop_duplicates()\n",
    "print(len(ref_gene_map_explode.index))\n",
    "print(len(ref_gene_map_explode['RefGene'].unique()))\n",
    "print(len(ref_gene_map_explode[ref_gene_map_explode['RefGene'].isna()].index))\n",
    "ref_gene_map_explode.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f73517e-5399-4b7c-b194-a0bb6a782653",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "21",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_sub = ref_gene_map %>%\n",
    "dplyr::select(TargetID,RefGene,Accession)\n",
    "head(ref_gene_map_sub)"
=======
   "id": "269c5e8f-c01d-4a4e-bb80-331f17b9f582",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_sub = ref_gene_map_explode[['TargetID', 'RefGene', 'Accession']]\n",
    "ref_gene_map_sub.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "22",
   "metadata": {},
   "source": [
    "## split rows with multiple genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode = ref_gene_map_sub %>%\n",
    "separate_rows(RefGene, Accession, sep = \";\")\n",
    "nrow(ref_gene_map_explode)\n",
    "length(unique(ref_gene_map_explode$RefGene))\n",
    "length(unique(ref_gene_map_explode$Accession))\n",
    "length(unique(ref_gene_map_explode$TargetID))\n",
    "head(ref_gene_map_explode)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode %>%\n",
    "filter(grepl('cg00001583', TargetID))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25",
=======
   "id": "7e85ea98-11f7-46a9-9a14-749685435eaf",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## change month/day genes"
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "26",
=======
   "id": "f6ec70b1-73f3-436f-ae93-ca1930318a4b",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "### filter to date genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "27",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode %>%\n",
    "arrange(RefGene) %>%\n",
    "distinct(RefGene) %>%\n",
    "head(n = 20)"
=======
   "id": "34203fbb-19bf-4d22-87cb-d88ffcb739df",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_sub.sort_values(by = 'RefGene').drop_duplicates(subset = 'RefGene').head(20)"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "28",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode_date = ref_gene_map_explode %>%\n",
    "filter(grepl('Dec|Mar|Sep', RefGene))\n",
    "unique(ref_gene_map_explode_date$RefGene)\n",
    "head(ref_gene_map_explode_date)"
=======
   "id": "79d4daf8-0fee-4c2b-98f6-9cd6afda73fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_date = ref_gene_map_sub[ref_gene_map_sub['RefGene'].str.contains('Dec|Mar|Sep')]\n",
    "ref_date['RefGene'].unique()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "29",
=======
   "id": "c23b5ea5-8bd0-424d-9d66-2b62d2323228",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## create map"
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "30",
=======
   "id": "5f10b97e-055c-465d-821d-2f38aca58de3",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "### create vectors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "31",
   "metadata": {},
   "outputs": [],
   "source": [
    "date_accession_vec = as.vector(unique(ref_gene_map_explode_date$Accession))\n",
    "date_accession_vec"
=======
   "id": "c5913ca3-f6bb-4584-8fc6-455331d6ecb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "date_accession_list = ref_date['Accession'].unique().tolist()\n",
    "date_accession_list"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "32",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_vec = c('MARCHF4',\n",
=======
   "id": "1a2a9b49-499a-47d6-a945-8c77a375a6d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = ['MARCHF4',\n",
>>>>>>> 55d97b9 (updated scripts)
    "             'MARCHF7',\n",
    "             'MARCHF1',\n",
    "             'SEPTIN11',\n",
    "             'MARCHF11',\n",
    "             'MARCHF3',\n",
    "             'MARCHF6',\n",
    "             'SEPTIN8',\n",
    "             'SEPTIN8',\n",
    "             'SEPTIN7P2',\n",
    "             'SEPTIN14',\n",
    "             'DELEC1',\n",
    "             'MARCHF5',\n",
<<<<<<< HEAD
    "            'MARCHF9',\n",
    "            'SEPTIN1',\n",
    "            'SEPTIN9',\n",
    "            'MARCHF10',\n",
    "            'SEPTIN5')\n",
    "gene_vec"
=======
    "             'MARCHF9',\n",
    "             'SEPTIN1',\n",
    "             'SEPTIN9',\n",
    "             'MARCHF10',\n",
    "             'SEPTIN5']\n",
    "gene_list"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "33",
=======
   "id": "153e3ace-3b77-4c9a-bb53-0dfa8c5de4c7",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "### create df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "34",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_accession_gene_map = data.frame(date_accession_vec,gene_vec)\n",
    "all_accession_gene_map=all_accession_gene_map %>%\n",
    "rename('Accession' = 'date_accession_vec')\n",
=======
   "id": "5257b1b4-fe81-4dd9-b8cf-505e0c4c5223",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_accession_gene_map = pd.DataFrame({'Accession' : date_accession_list, 'Gene' : gene_list})\n",
>>>>>>> 55d97b9 (updated scripts)
    "all_accession_gene_map"
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "35",
=======
   "id": "069783a6-025b-4c8b-9990-dff301491bac",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "36",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode_date_correct_genes = ref_gene_map_explode_date %>%\n",
    "inner_join(all_accession_gene_map, by='Accession')\n",
    "nrow(ref_gene_map_explode_date_correct_genes)\n",
    "nrow(ref_gene_map_explode_date_correct_genes)\n",
    "head(ref_gene_map_explode_date_correct_genes)"
=======
   "id": "b9d19dfa-476f-4350-8783-33420f07a2cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_date_correct = ref_date.merge(all_accession_gene_map, on = 'Accession')\n",
    "print(len(ref_date_correct.index))\n",
    "ref_date_correct.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "37",
=======
   "id": "0b8baaf1-4751-4405-8b32-43cc63f1ef55",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "38",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode_date_correct_genes = ref_gene_map_explode_date_correct_genes %>%\n",
    "dplyr::select(TargetID, gene_vec, Accession) %>%\n",
    "rename('RefGene' = 'gene_vec')\n",
    "head(ref_gene_map_explode_date_correct_genes)"
=======
   "id": "4194afcd-174a-4852-8fd4-1656772c77ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_date_correct = ref_date_correct[['TargetID', 'Gene', 'Accession']].rename(columns = {'Gene' : 'RefGene'})\n",
    "ref_date_correct.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "39",
=======
   "id": "b3bbce23-6937-4375-ae77-026bb5c5de5c",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## create df without date genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "40",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode_no_date = ref_gene_map_explode %>%\n",
    "filter(!grepl('Dec|Mar|Sep', RefGene))\n",
    "head(ref_gene_map_explode_no_date)"
=======
   "id": "ef78c25d-4478-48bc-9ec6-5f128b375f9f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ref_no_date = ref_gene_map_sub[~ref_gene_map_sub.index.isin(ref_date.index)]\n",
    "print(len(ref_gene_map_sub.index))\n",
    "print(len(ref_date.index))\n",
    "print(len(ref_no_date.index))"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "41",
=======
   "id": "d3256016-b48a-4763-a3f3-4c6e8efefefd",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## concatenate date + fixed date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "42",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_explode_fixed = ref_gene_map_explode_no_date %>%\n",
    "rbind(ref_gene_map_explode_date_correct_genes)\n",
    "nrow(ref_gene_map_explode_fixed)\n",
    "nrow(ref_gene_map_explode)\n",
    "head(ref_gene_map_explode_fixed)"
=======
   "id": "5f4f70b8-131c-4200-836b-9c6f08afea8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_fixed = pd.concat([ref_no_date, ref_date_correct], axis = 0)\n",
    "print(len(ref_gene_map_fixed.index))\n",
    "print(len(ref_gene_map_sub.index))"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "43",
=======
   "id": "d9eb5c91-eb22-4d8c-a272-53635e6a030d",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## clean up ref seq file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "44",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_direct = ref %>%\n",
    "filter(info_type == 'DIRECT')\n",
    "nrow(ref)\n",
    "nrow(ref_direct)\n",
    "head(ref_direct)"
=======
   "id": "1e6fc4fb-1b83-44f8-91ef-3d9723eef259",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_direct = ref[ref['info_type'] == 'DIRECT']\n",
    "print(len(ref.index))\n",
    "print(len(ref_direct.index))\n",
    "ref_direct['info_type'].unique()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "45",
=======
   "id": "f5117fdb-0a91-4625-b1e2-614f4fe2e1f8",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## filter gene map to ref transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "46",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_ref_trans = ref_gene_map_explode_fixed[ref_gene_map_explode_fixed$Accession %in% ref_direct$xref,]\n",
    "nrow(ref_gene_map_explode_fixed)\n",
    "nrow(ref_gene_map_ref_trans)\n",
    "length(unique(ref_gene_map_ref_trans$Accession))\n",
    "head(ref_gene_map_ref_trans)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(unique(ref_gene_map_ref_trans$RefGene))\n",
    "length(unique(ref_gene_map_ref_trans$TargetID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48",
   "metadata": {},
   "outputs": [],
   "source": [
    "dups <- ref_gene_map_ref_trans[duplicated(ref_gene_map_ref_trans$TargetID) | duplicated(ref_gene_map_ref_trans$TargetID, fromLast = TRUE), ]\n",
    "head(dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_ref_trans_no_dup = ref_gene_map_ref_trans %>%\n",
    "distinct()\n",
    "nrow(ref_gene_map_ref_trans)\n",
    "nrow(ref_gene_map_ref_trans_no_dup)\n",
    "head(ref_gene_map_ref_trans_no_dup)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_ref_trans_no_dup_again = ref_gene_map_ref_trans_no_dup %>%\n",
    "distinct(TargetID, RefGene, .keep_all = T)\n",
    "nrow(ref_gene_map_ref_trans_no_dup)\n",
    "nrow(ref_gene_map_ref_trans_no_dup_again)\n",
    "head(ref_gene_map_ref_trans_no_dup_again)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51",
   "metadata": {},
   "outputs": [],
   "source": [
    "still_dups <- ref_gene_map_ref_trans_no_dup_again[duplicated(ref_gene_map_ref_trans_no_dup_again$TargetID) | duplicated(ref_gene_map_ref_trans_no_dup_again$TargetID, fromLast = TRUE), ]\n",
    "head(still_dups)"
=======
   "id": "4bbbd5b7-3c2b-47ba-bcd1-5aa302a41178",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_gene_map_ref_trans = ref_gene_map_fixed[ref_gene_map_fixed['Accession'].isin(ref_direct['xref'])]\n",
    "print(len(ref_gene_map_ref_trans.index))\n",
    "print(len(ref_gene_map_fixed.index))"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "52",
   "metadata": {},
   "source": [
    "## create unique gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = ref_gene_map_ref_trans_no_dup_again %>%\n",
    "dplyr::select(RefGene) %>%\n",
    "distinct()\n",
    "nrow(gene_list)\n",
    "head(gene_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54",
=======
   "id": "a1061800-1c5c-4aef-a01c-7007937c541f",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "# filter methylation data to ref genes"
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "55",
=======
   "id": "63f28c9b-2264-4235-82a7-81666397cc7d",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "56",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref = ref_gene_map_ref_trans_no_dup_again %>%\n",
    "inner_join(methylation, by = 'TargetID')\n",
    "nrow(methyl_ref)\n",
    "length(unique(methyl_ref$TargetID))\n",
    "length(unique(methyl_ref$RefGene))\n",
    "head(methyl_ref)"
=======
   "id": "c9e95113-78c5-4a00-aa76-386bcec338ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref = ref_gene_map_ref_trans.merge(methylation, on = 'TargetID')\n",
    "print(len(methyl_ref.index))\n",
    "print(len(methylation.index))\n",
    "methyl_ref.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "57",
   "metadata": {},
   "source": [
    "## create column combining RefGene and TargetID"
=======
   "id": "27a78226-b4e8-4e4b-b5c0-3b2bbe934839",
   "metadata": {},
   "source": [
    "# calculate weight\n",
    "- 1 / # genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33c2a657-9635-4d86-91f3-03294460f344",
   "metadata": {},
   "source": [
    "## get counts per gene"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "58",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_id = methyl_ref %>%\n",
    "mutate(RefGene_TargetID_Accession = paste0(RefGene, '_', TargetID, '_', Accession)) %>%\n",
    "select(RefGene_TargetID_Accession) %>%\n",
    "cbind(methyl_ref) %>%\n",
    "select(-c(TargetID, RefGene, Accession)) %>%\n",
    "distinct()\n",
    "nrow(methyl_ref_id)\n",
    "length(unique(methyl_ref_id$RefGene_TargetID_Accession))\n",
    "head(methyl_ref_id)"
=======
   "id": "ce78634e-e2a7-48b8-9bdb-06d5d9b68e5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count = pd.DataFrame(methyl_ref[['TargetID', 'RefGene']].drop_duplicates()['TargetID'].value_counts()).reset_index()\n",
    "print(gene_count['count'].describe())\n",
    "gene_count.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "404dbaaa-92dc-4ca7-aa1b-293175e13aaf",
   "metadata": {},
   "source": [
    "## merge and calculate weight"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "59",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_id%>%\n",
    "filter(duplicated(RefGene_TargetID_Accession) | duplicated(RefGene_TargetID_Accession, fromLast = TRUE)) %>%\n",
    "arrange(RefGene_TargetID_Accession) %>%\n",
    "head()"
=======
   "id": "336c920f-eef1-4fd6-9f31-7c38386ebebd",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_gene_weight = methyl_ref.merge(gene_count, on = 'TargetID')\n",
    "methyl_gene_weight['weight'] = 1/methyl_gene_weight['count']\n",
    "print(len(methyl_ref.index))\n",
    "print(len(methyl_gene_weight.index))\n",
    "methyl_gene_weight.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "60",
=======
   "id": "c033606f-f863-449b-9bce-639549bd8c7d",
   "metadata": {},
   "source": [
    "# convert to m-values"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90ce2b3e-dc37-45d5-9d40-1ab3543387a2",
   "metadata": {},
   "source": [
    "## create function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f957331-ea06-4cd9-85d9-0d93f69a9674",
   "metadata": {},
   "outputs": [],
   "source": [
    "def beta_to_m(beta):\n",
    "    beta = beta.copy()\n",
    "    beta[(beta <= 0) | (beta >= 1)] = np.nan\n",
    "    return np.log2(beta / (1 - beta))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a2f63cf-65d9-4708-9a44-29dc6c8e4431",
   "metadata": {},
   "source": [
    "## extract id, gene, accession, weight, and count cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6d54fb0-0133-4b38-aa40-4d8fbd9720dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_id_only = methyl_gene_weight[['TargetID', 'RefGene', 'Accession', 'count', 'weight']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb0b7a23-7617-4342-80f7-8d3978892ded",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_no_id = methyl_gene_weight.drop(columns = ['TargetID', 'RefGene', 'Accession', 'count', 'weight'])\n",
    "methyl_no_id.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47d4d46a-0786-45d1-a406-8b0b2406738a",
   "metadata": {},
   "source": [
    "## convert to float"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d1db509-ab64-4c62-a243-2e49518d4030",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_no_id = methyl_no_id.astype(float)\n",
    "methyl_no_id.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca81628b-4da0-4217-b821-c5cc91b6068c",
   "metadata": {},
   "source": [
    "## convert to m-values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c12be76c-e7ab-4f3b-9f49-93d4994632a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_norm = beta_to_m(methyl_no_id)\n",
    "methyl_norm.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c477c60c-4e8a-40e9-a78d-42b071921aee",
   "metadata": {},
   "source": [
    "## add id cols back in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f355bd6-e5ed-4381-a51e-85dbbc7cf1cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_norm_id = pd.concat([methyl_id_only, methyl_norm], axis = 1)\n",
    "methyl_norm_id.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f177b38-e993-4661-8742-3d8cbc82c21f",
   "metadata": {},
   "source": [
    "## replace inf with na"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c80466f8-87ea-476d-ab8e-46984b6015e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_norm_id = methyl_norm_id.replace([np.inf, -np.inf], np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "274667d8-0e00-4449-9f41-eb51cf1d5782",
   "metadata": {},
   "source": [
    "# compute weighted average by gene"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2cc1208-0f54-446d-af99-8239468f0e30",
   "metadata": {},
   "source": [
    "## define function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd80a5f0-6695-4e4e-bc4b-7cb02e47b010",
   "metadata": {},
   "outputs": [],
   "source": [
    "def weighted_average(values, weights):\n",
    "    mask = values.notna() & weights.notna()\n",
    "    if weights[mask].sum() == 0:\n",
    "        return np.nan\n",
    "    else:\n",
    "        return (values * weights).sum(skipna = True) / weights[mask].sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c552aa60-97f0-4aa0-ac6a-b5e080b7b48b",
   "metadata": {},
   "source": [
    "## compute"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d052f085-0f7b-4c25-9f51-579f2afd17a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "avg_columns = methyl_norm_id.columns.tolist()\n",
    "avg_columns.remove('TargetID')\n",
    "avg_columns.remove('Accession')\n",
    "avg_columns.remove('RefGene')\n",
    "avg_columns.remove('weight')\n",
    "avg_columns.remove('count')\n",
    "\n",
    "methyl_avg = (methyl_norm_id.groupby('RefGene').apply(lambda g: pd.Series({col: weighted_average(g[col], g['weight']) for col in avg_columns})).reset_index())\n",
    "print(methyl_norm_id.shape)\n",
    "print(methyl_avg.shape)\n",
    "print(len(methyl_avg['RefGene'].unique()))\n",
    "methyl_avg.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56a7a3c8-a1ed-446d-8a05-162c2f7ec567",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(methyl_norm_id.shape)\n",
    "print(methyl_avg.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edebadd0-13d3-4b36-9651-16a4261cad9c",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "# map to individual ids"
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "61",
=======
   "id": "1def6696-a88f-40f6-8804-45117c4b16ca",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## transpose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "62",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_tranpose = data.frame(t(methyl_ref_id))\n",
    "head(methyl_ref_tranpose)"
=======
   "id": "173ffadb-ef43-4597-9082-36ed43507e7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_transpose = methyl_avg.transpose()\n",
    "methyl_transpose.columns = methyl_transpose.iloc[0]\n",
    "methyl_transpose = methyl_transpose.iloc[1:]\n",
    "methyl_transpose['specimenID'] = methyl_transpose.index\n",
    "methyl_transpose.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "63",
   "metadata": {},
   "source": [
    "## rename columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(methyl_ref_tranpose) = as.character(methyl_ref_tranpose[1, ])\n",
    "head(methyl_ref_tranpose)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65",
   "metadata": {},
   "source": [
    "## drop first row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_tranpose <- methyl_ref_tranpose[-1, ]\n",
    "head(methyl_ref_tranpose)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "67",
   "metadata": {},
   "source": [
    "## add ID column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_tranpose_id = data.frame(rownames(methyl_ref_tranpose))\n",
    "colnames(methyl_ref_tranpose_id) = c('specimenID')\n",
    "head(methyl_ref_tranpose_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_tranpose = cbind(methyl_ref_tranpose_id, methyl_ref_tranpose)\n",
    "head(methyl_ref_tranpose)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70",
=======
   "id": "d8706428-e568-428c-89ba-1c202e6078d7",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "## add individual ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "71",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_tranpose_id_map = id_map %>%\n",
    "inner_join(methyl_ref_tranpose, by = 'specimenID')\n",
    "nrow(methyl_ref_tranpose_id_map)\n",
    "nrow(methyl_ref_tranpose)\n",
    "nrow(id_map)\n",
    "head(methyl_ref_tranpose_id_map)"
=======
   "id": "535d1d3e-70d7-4741-931d-c03e472a2662",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_transpose_id_map = id_map.merge(methyl_transpose, on = 'specimenID')\n",
    "print(len(methyl_transpose_id_map.index))\n",
    "print(len(methyl_transpose.index))\n",
    "print(len(id_map.index))\n",
    "methyl_transpose_id_map.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "72",
   "metadata": {},
   "source": [
    "## drop specimenID"
=======
   "id": "afe69eb6-e6f4-42a3-b8e8-73f57d017167",
   "metadata": {},
   "source": [
    "## add in common id"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "73",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_tranpose_id_map = methyl_ref_tranpose_id_map %>%\n",
    "dplyr::select(-c(specimenID))\n",
    "head(methyl_ref_tranpose_id_map)"
=======
   "id": "3b26dccd-0908-4a48-a58e-bf48e68d8914",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_transpose_common_id_map = common_id_map.merge(methyl_transpose_id_map, on = 'individualID')\n",
    "print(len(methyl_transpose_id_map.index))\n",
    "print(len(methyl_transpose_common_id_map.index))\n",
    "methyl_transpose_common_id_map.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "74",
   "metadata": {},
   "source": [
    "# convert to m-values (logit transformed beta values)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "75",
   "metadata": {},
   "source": [
    "formula:\n",
    "\n",
    "M = log2(beta / (1 - beta))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76",
   "metadata": {},
   "source": [
    "## drop id col"
=======
   "id": "16c23cf2-be05-4f0f-a700-a17ff1be4034",
   "metadata": {},
   "source": [
    "## drop other id cols"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "77",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_id = methyl_ref_tranpose_id_map %>%\n",
    "select(individualID)"
=======
   "id": "02d3e974-80e7-48bc-8eeb-ce567591b43f",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_transpose_common_id_map = methyl_transpose_common_id_map.drop(columns = ['SampleID', 'individualID', 'specimenID'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea80e43a-3e5c-4566-b424-6eaee8fc70b9",
   "metadata": {},
   "source": [
    "# transpose again"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "78",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_no_id = methyl_ref_tranpose_id_map %>%\n",
    "select(-c(individualID))"
=======
   "id": "c43dae51-d373-4771-871d-e3f23fa8c57f",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_norm_reformat = methyl_transpose_common_id_map.set_index('CommonID')\n",
    "methyl_norm_reformat = methyl_norm_reformat.transpose()\n",
    "methyl_norm_reformat.insert(0, 'GENE', methyl_norm_reformat.index)\n",
    "methyl_norm_reformat.head()"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "79",
   "metadata": {},
   "source": [
    "## convert to numeric"
=======
   "id": "56cb0db7-eefe-43c9-b057-237c6ad518a2",
   "metadata": {},
   "source": [
    "# make gene list"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "80",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_no_id[] = lapply(methyl_no_id, function(x) as.numeric(as.character(x)))\n",
    "head(methyl_no_id)"
=======
   "id": "9f69a5a0-4cbe-44e8-8ab5-b5a515c8bfcf",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = methyl_norm_reformat[['GENE']]"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "markdown",
<<<<<<< HEAD
   "id": "81",
   "metadata": {},
   "source": [
    "## create normalizing function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82",
   "metadata": {},
   "outputs": [],
   "source": [
    "beta_to_m <- function(beta) {\n",
    "  log2(beta / (1 - beta))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83",
   "metadata": {},
   "source": [
    "## normalize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_norm <- as.data.frame(lapply(methyl_no_id, beta_to_m))\n",
    "head(methyl_ref_norm)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85",
   "metadata": {},
   "source": [
    "## add id back in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_ref_norm_id = cbind(methyl_id, methyl_ref_norm)\n",
    "head(methyl_ref_norm_id)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87",
=======
   "id": "32257724-38a3-4187-9b16-85f10d91bfb1",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "source": [
    "# export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "88",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(gene_list,\n",
    "            'rosmap/ROSMAP_arrayMethylation_imputed.gene_list.txt',\n",
    "            sep = '\\t',\n",
    "            col.names = T,\n",
    "            row.names = F,\n",
    "            quote = F)"
=======
   "id": "e22119fe-fbb5-490f-b4b3-527f325f60d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list.to_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.weighted_gene_average.gene_list.txt',\n",
    "                 sep = '\\t',\n",
    "                 index = None)"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "89",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(methyl_ref_norm_id,\n",
    "            'rosmap/ROSMAP_arrayMethylation_imputed.gene_symbol.individualID.mvalue_norm.txt',\n",
    "            sep = '\\t',\n",
    "            col.names = T,\n",
    "            row.names = F,\n",
    "            quote = F)"
=======
   "id": "3ca85f55-dcd0-4b05-8ee5-daded16256ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "methyl_norm_reformat.to_csv('pathway_score/rosmap/ROSMAP_arrayMethylation_imputed.gene_symbol.individualID.mvalue_norm.weighted_gene_average.common_id.transpose.txt',\n",
    "                            sep = '\\t',\n",
    "                            index = None)"
>>>>>>> 55d97b9 (updated scripts)
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
<<<<<<< HEAD
   "id": "90",
=======
   "id": "4d349c17-95f3-4eec-8842-3f367b196709",
>>>>>>> 55d97b9 (updated scripts)
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
<<<<<<< HEAD
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.3"
=======
   "display_name": "Python 3.12 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
>>>>>>> 55d97b9 (updated scripts)
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
